dockstore-webservice/src/main/java/io/dockstore/webservice/resources/LambdaEventResource.java

```diff
@@ -62,11 +65,12 @@ public LambdaEventResource(SessionFactory sessionFactory) {
             @ApiParam(value = PAGINATION_OFFSET_TEXT) @QueryParam("offset") @DefaultValue("0") String offset,
             @ApiParam(value = PAGINATION_LIMIT_TEXT, allowableValues = "range[1,100]", defaultValue = PAGINATION_LIMIT) @DefaultValue(PAGINATION_LIMIT) @QueryParam("limit") Integer limit) {
         // To ensure a user has access to an organization, check that they have at least one workflow from that organization
-        List<Workflow> workflows = workflowDAO.findMyEntries(user.getId());
-        boolean canAccessOrganization = workflows.stream().anyMatch(workflow -> Objects.equals(workflow.getOrganization(), organization) && Objects.equals(workflow.getSourceControl(),
+        User authUser = userDAO.findById(user.getId());
+        List<Workflow> workflows = workflowDAO.findByOrganization(organization);
```

Comment:
This cuts down data loaded in memory from all workflows to an organization's workflows, which is a lot better. 
But can still be better by avoiding loading workflows at all since checking that the user has access can be done purely by query or checking with GitHub that the user had access to the organization directly in the case that the user has not been added to a workflow in the org yet would be more efficient. 
Can be added as a code comment (TODO) and ticket

Reply:
I ended up just changing it to call the GitHub API, much cleaner now.